sdk/typescript/scripts/generate.ts

```diff
@@ -0,0 +1,545 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import ts from 'typescript';
+import { format } from 'prettier';
+import * as fs from 'node:fs/promises';
+import * as path from 'node:path';
+import { OpenRpcType, type OpenRpcSpec, OpenRpcTypeRef } from './open-rpc';
+/** @ts-ignore */
+import prettierConfig from '../../../prettier.config.js';
+
+const packageRoot = path.resolve(import.meta.url.slice(5), '../..');
+const openRpcSpec: OpenRpcSpec = JSON.parse(
+	await fs.readFile(
+		path.resolve(packageRoot, '../../crates/sui-open-rpc/spec/openrpc.json'),
+		'utf-8',
+	),
+);
+const LICENSE_HEADER = `
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+`.trim();
+
+const options: {
+	types: Record<
+		string,
+		{
+			typeAlias?: string;
+			alias?: string;
+			flatten?: boolean;
+		}
+	>;
+} = {
+	types: {
```

Comment:
The type names we used in the SDK had a lot of differences from what we use the the rpc spec.  these aliases ensure that all the types we use in the sui repo are mapped to the names we were using before.  I don't love this, and could easily be convinced we should just have the renames be part of the big set of changes we are releasing.  That being said, the names we use in the rpc spec are all over the place too, wouldn't be my first choice if I were starting from scratch.